tests/integration/customerc20.test.ts

```diff
@@ -219,6 +214,36 @@ const registerCustomToken = async (
     'Start l2Erc20Address not equal empty address'
   ).to.eq(constants.AddressZero)
 
+  // it should fail without the approval
+  if (isL2NetworkWithCustomFeeToken()) {
+    try {
+      const regTx = await adminErc20Bridger.registerCustomToken(
+        l1CustomToken.address,
+        l2CustomToken.address,
+        l1Signer,
+        l2Signer.provider!
+      )
+      await regTx.wait()
+      throw new Error('L2 custom token is not approved but got deployed')
+    } catch (err) {
+      expect((err as Error).message).to.contain('Insufficient allowance')
+    }
+  }
+
+  const gEstimator = new L1ToL2MessageGasEstimator(l2Signer.provider!)
+  // increase by 500%
+  const maxFeePerGas = (await gEstimator.estimateMaxFeePerGas()).mul(6)
+  // hardcode gas limit to 60k
+  const estimatedGas = BigNumber.from(60_000).mul(maxFeePerGas)
+
+  if (isL2NetworkWithCustomFeeToken()) {
+    const approvalTx = await ERC20__factory.connect(
```

Comment:
This code should be inside `approveGasToken` and `getApproveGasTokenRequest` of `AdminErc20Bridger`